src/Formlet/Ocelot/Checkbox.purs

```diff
@@ -13,16 +13,20 @@ import Data.Set as Data.Set
 import Formlet as Formlet
 import Formlet.Ocelot.Enum as Formlet.Ocelot.Enum
 import Formlet.Render as Formlet.Render
+import Option as Option
+import Record as Record
 
 newtype Render a =
   Render
-    ( Array
-        { checked :: Boolean
-        , label :: String
-        , onChange :: Boolean -> a
-        , readonly :: Boolean
-        }
-    )
+    { columns :: Array Int -- NOTE positive integer (> 0), we use Array for its Semigroup instance and to collect column values
```

Comment:
I think we can let the `render` function decide to use `max` if we want that behavior.

Reply:
The number of items in the `Array Int` depends on the number of times we use `Semigroup` instance to combine subforms which doesn't make much sense to me.

Reply:
> I think we can let the `render` function decide to use `max` if we want that behavior.

The semantics should lie in the DSL layer not in the Halogen implementation.

Reply:
Well then I guess we need to decide what behavior we want. An argument could be made that `Data.Semigroup.First` is reasonable.

@davezuch do you have an opinion?

Reply:
Either is fine with me as long as we document the behavior.

Reply:
Wouldn't it make more sense to define a `RenderSet` type in this module with this `columns` property (do all `Render` types need to implement `Semigroup` or is it just a constraint of the checkbox `Render`?) and use that for `checkboxSet` instead of `Formlet.Render.Render`? https://github.com/citizennet/purescript-formlet-ocelot/blob/c09a5b746646241c33ca7da64ba21e6ca83f041e/src/Formlet/Ocelot/Checkbox.purs#L83C62-L83C62

Reply:
> Wouldn't it make more sense to define a `RenderSet` type in this module with this `columns` property (do all `Render` types need to implement `Semigroup` or is it just a constraint of the checkbox `Render`?) and use that for `checkboxSet` instead of `Formlet.Render.Render`? https://github.com/citizennet/purescript-formlet-ocelot/blob/c09a5b746646241c33ca7da64ba21e6ca83f041e/src/Formlet/Ocelot/Checkbox.purs#L83C62-L83C62

If we do this I think it makes sense to move `checkboxSet` to its own module. We're going to have to add another option for to `Formlet.Ocelot.Render.Renders` anyway.

Reply:
> If we do this I think it makes sense to move `checkboxSet` to its own module. We're going to have to add another option for it to `Formlet.Ocelot.Render.Renders` anyway.

That sounds like re-engineering of the whole checkbox `Render` which might be an overkill for what we want to achieve.

Reply:
> > If we do this I think it makes sense to move `checkboxSet` to its own module. We're going to have to add another option for it to `Formlet.Ocelot.Render.Renders` anyway.
> 
> That sounds like re-engineering of the whole checkbox `Render` which might be an overkill for what we want to achieve.

`checkbox` and `checkboxSet` are only superficially similar. Their value and result types are different. In some ways, `checkboxSet` behaves more like a typeahead.

Reply:
I'm going to try splitting them apart and see where it takes me.